pkg/queue/manager.go

```diff
@@ -343,15 +343,21 @@ func (m *Manager) RequeueWorkload(ctx context.Context, info *workload.Info, reas
 	// Always get the newest workload to avoid requeuing the out-of-date obj.
 	err := m.client.Get(ctx, client.ObjectKeyFromObject(info.Obj), &w)
 	// Since the client is cached, the only possible error is NotFound
-	if apierrors.IsNotFound(err) || workload.HasQuotaReservation(&w) {
+	if apierrors.IsNotFound(err) {
+		return false
+	}
+
+	info.Update(&w)
```

Comment:
We actually don't want to update the object in this case.

If the workload has a reservation, we don't want to unset that reservation in `requeueAndUpdate`. The patch should fail, because the ResourceVersion won't match, and that's what we want.

Reply:
Another alternative would be to update the object, but then check again in `requeueAndUpdate` whether the object still doesn't have a quota reservation.

Reply:
Reverted